nautobot/ipam/models.py

```diff
@@ -1295,33 +1296,61 @@ class VLANGroup(OrganizationalModel):
     )
     description = models.CharField(max_length=CHARFIELD_MAX_LENGTH, blank=True)
 
+    range = PositiveRangeNumberTextField(
+        blank=False, default="1-4094", help_text="Vlan range, default 1-4094 if left blank."
+    )
+
     class Meta:
         ordering = ("name",)
         verbose_name = "VLAN group"
         verbose_name_plural = "VLAN groups"
 
+    @property
+    def available_vids(self):
+        _group_expanded_vids = parse_numeric_range(self.range)
+        used_ids = self.vlans.all().values_list("vid", flat=True)
+        available = sorted([vid for vid in _group_expanded_vids if vid not in used_ids])
+
+        return available
+
     def clean(self):
         super().clean()
-
         # Validate location
         if self.location is not None:
             if ContentType.objects.get_for_model(self) not in self.location.location_type.content_types.all():
                 raise ValidationError(
                     {"location": f'VLAN groups may not associate to locations of type "{self.location.location_type}".'}
                 )
 
+        # Get expanded range from Field's value
+        _expanded_range = parse_numeric_range(self.range)
+
+        # Validate range boundaries
+        if _expanded_range[0] < constants.VLAN_VID_MIN or _expanded_range[-1] > constants.VLAN_VID_MAX:
+            raise ValidationError(
+                {
+                    "range": f"VLAN group range is outside of allowed {constants.VLAN_VID_MIN}-{constants.VLAN_VID_MAX} "
+                    f"range."
+                }
+            )
```

Comment:
This check feels like it could be implemented directly in the `PositiveRangeNumberTextField` class perhaps, or done with an appropriate Django `validator` implementation?

Reply:
Moved to `validate` method under `PositiveRangeNumberTextField`
Implemented min_boundary and max_boundary on `PositiveRangeNumberTextField`